Read panicked with nil pointer dereference

Hello everyone,
after upgrading from v0.5.3 to v0.7, the following panic occurred after a series of failed read/write attempts (EOF, StatusBadTimeout). The laptop was in stand-by at that time and continuously losing its WIFI connection.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x68 pc=0xea0489]


goroutine 104 [running]:
github.com/gopcua/opcua/uasc.(*SecureChannel).SendRequestWithTimeout(0x0, {0x1aad1d0, 0xc000446700}, {0x1aa7fc0, 0xc001212ba0}, 0xc000d72540, 0x2540be400, 0xc0012c7860)
	/go/pkg/mod/github.com/gopcua/opcua@v0.7.0/uasc/secure_channel.go:979 +0x49
github.com/gopcua/opcua.(*Client).sendWithTimeout(0x25ed1a0?, {0x1aad1d0?, 0xc000446700?}, {0x1aa7fc0?, 0xc001212ba0?}, 0x41bd65?, 0x8?)
	/go/pkg/mod/github.com/gopcua/opcua@v0.7.0/client.go:959 +0x132
github.com/gopcua/opcua.(*Client).Send(0xc00074c0e0, {0x1aad1d0, 0xc000446700}, {0x1aa7fc0, 0xc001212ba0}, 0xc0012c7860)
	/go/pkg/mod/github.com/gopcua/opcua@v0.7.0/client.go:941 +0x88
github.com/gopcua/opcua.(*Client).Read(0xc00074c0e0, {0x1aad1d0, 0xc000446700}, 0xc0012128d0)
	/go/pkg/mod/github.com/gopcua/opcua@v0.7.0/client.go:1048 +0x28c
****************************/pkg/nuopc.Read({0x1aad1d0, 0xc000446700}, 0xc00074c0e0, {0x15ef2e0?, 0xc001286900?}, 0xc0012c7b88?)
	/app/pkg/nuopc/opc.go:36 +0x268
****************************/core/plc.(*readQO).Execute(0xc001284e10, 0xc00074c0e0)
	/app/core/plc/queue_object.go:71 +0x93
****************************/core/plc.(*PLC).run(0xc0008b2840, 0xc00090fbe0)
	/app/core/plc/plc.go:402 +0x4b2
****************************/core/plc.InitShared.func1 in goroutine 1
	/app/core/plc/plc.go:66 +0x2e7
```

This has only happened once since the update, and I couldn't manage to recreate it (tried by repeatedly turning WIFI off/on). Any help would be much appreciated.

Thanks in advance and best regards.

--- a/.github/workflows/go.yml
+++ b/.github/workflows/go.yml
@@ -15,7 +15,7 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, windows-latest]
-        go: ["1.22.x", "1.23.x"]
+        go: ["1.23.x", "1.24.x"]
 
     steps:
       - name: Check out code into the Go module directory

--- a/client.go
+++ b/client.go
@@ -736,7 +736,8 @@ func (s *Session) RevisedTimeout() time.Duration {
 //
 // See Part 4, 5.6.2
 func (c *Client) CreateSession(ctx context.Context, cfg *uasc.SessionConfig) (*Session, error) {
-	if c.SecureChannel() == nil {
+	sc := c.SecureChannel()
+	if sc == nil {
 		return nil, ua.StatusBadServerNotConnected
 	}
 
@@ -761,14 +762,14 @@ func (c *Client) CreateSession(ctx context.Context, cfg *uasc.SessionConfig) (*S
 
 	var s *Session
 	// for the CreateSessionRequest the authToken is always nil.
-	// use c.SecureChannel().SendRequest() to enforce this.
-	err := c.SecureChannel().SendRequest(ctx, req, nil, func(v ua.Response) error {
+	// use sc.SendRequest() to enforce this.
+	err := sc.SendRequest(ctx, req, nil, func(v ua.Response) error {
 		var res *ua.CreateSessionResponse
 		if err := safeAssign(v, &res); err != nil {
 			return err
 		}
 
-		err := c.SecureChannel().VerifySessionSignature(res.ServerCertificate, nonce, res.ServerSignature.Signature)
+		err := sc.VerifySessionSignature(res.ServerCertificate, nonce, res.ServerSignature.Signature)
 		if err != nil {
 			log.Printf("error verifying session signature: %s", err)
 			return nil
@@ -827,11 +828,12 @@ func anonymousPolicyID(endpoints []*ua.EndpointDescription) string {
 //
 // See Part 4, 5.6.3
 func (c *Client) ActivateSession(ctx context.Context, s *Session) error {
-	if c.SecureChannel() == nil {
+	sc := c.SecureChannel()
+	if sc == nil {
 		return ua.StatusBadServerNotConnected
 	}
 	stats.Client().Add("ActivateSession", 1)
-	sig, sigAlg, err := c.SecureChannel().NewSessionSignature(s.serverCertificate, s.serverNonce)
+	sig, sigAlg, err := sc.NewSessionSignature(s.serverCertificate, s.serverNonce)
 	if err != nil {
 		log.Printf("error creating session signature: %s", err)
 		return nil
@@ -842,7 +844,7 @@ func (c *Client) ActivateSession(ctx context.Context, s *Session) error {
 		// nothing to do
 
 	case *ua.UserNameIdentityToken:
-		pass, passAlg, err := c.SecureChannel().EncryptUserPassword(s.cfg.AuthPolicyURI, s.cfg.AuthPassword, s.serverCertificate, s.serverNonce)
+		pass, passAlg, err := sc.EncryptUserPassword(s.cfg.AuthPolicyURI, s.cfg.AuthPassword, s.serverCertificate, s.serverNonce)
 		if err != nil {
 			log.Printf("error encrypting user password: %s", err)
 			return err
@@ -851,7 +853,7 @@ func (c *Client) ActivateSession(ctx context.Context, s *Session) error {
 		tok.EncryptionAlgorithm = passAlg
 
 	case *ua.X509IdentityToken:
-		tokSig, tokSigAlg, err := c.SecureChannel().NewUserTokenSignature(s.cfg.AuthPolicyURI, s.serverCertificate, s.serverNonce)
+		tokSig, tokSigAlg, err := sc.NewUserTokenSignature(s.cfg.AuthPolicyURI, s.serverCertificate, s.serverNonce)
 		if err != nil {
 			log.Printf("error creating session signature: %s", err)
 			return err
@@ -875,7 +877,7 @@ func (c *Client) ActivateSession(ctx context.Context, s *Session) error {
 		UserIdentityToken:          ua.NewExtensionObject(s.cfg.UserIdentityToken),
 		UserTokenSignature:         s.cfg.UserTokenSignature,
 	}
-	return c.SecureChannel().SendRequest(ctx, req, s.resp.AuthenticationToken, func(v ua.Response) error {
+	return sc.SendRequest(ctx, req, s.resp.AuthenticationToken, func(v ua.Response) error {
 		var res *ua.ActivateSessionResponse
 		if err := safeAssign(v, &res); err != nil {
 			return err
@@ -956,7 +958,7 @@ func (c *Client) sendWithTimeout(ctx context.Context, req ua.Request, timeout ti
 	if s := c.Session(); s != nil {
 		authToken = s.resp.AuthenticationToken
 	}
-	return c.SecureChannel().SendRequestWithTimeout(ctx, req, authToken, timeout, h)
+	return sc.SendRequestWithTimeout(ctx, req, authToken, timeout, h)
 }
 
 // Node returns a node object which accesses its attributes

--- a/client_sub.go
+++ b/client_sub.go
@@ -181,7 +181,7 @@ func (c *Client) sendRepublishRequests(ctx context.Context, sub *Subscription, a
 
 		debug.Printf("RepublishRequest: req=%s", debug.ToJSON(req))
 		var res *ua.RepublishResponse
-		err := c.SecureChannel().SendRequest(ctx, req, c.Session().resp.AuthenticationToken, func(v ua.Response) error {
+		err := sc.SendRequest(ctx, req, c.Session().resp.AuthenticationToken, func(v ua.Response) error {
 			return safeAssign(v, &res)
 		})
 		debug.Printf("RepublishResponse: res=%s err=%v", debug.ToJSON(res), err)

